src/relay_garbo.ts

```diff
@@ -0,0 +1,54 @@
+import { fileToArray, formField, formFields, svnInfo, writeln } from "kolmafia";
+import { get, set } from "libram";
+
+export function main(): void {
+  // handle updating values
+  const fields = formFields();
+  Object.keys(fields).forEach((field) => {
+    if (field.includes("_didchange")) return;
+    if (field === "relay") return;
+
+    const oldSetting = formField(`${field} _didchange`);
+    if (oldSetting === fields[field]) {
+      if (get(field) !== fields[field]) {
+        writeln(
+          `You did not change setting ${field}. It changed since you last loaded the page, ignoring.<br>`
+        );
+      }
+      return;
+    }
+    if (get(field) !== fields[field]) {
+      writeln(`Changing setting ${field} to ${fields[field]} <br>`);
+      set(field, fields[field]);
+    }
+  });
+
+  // load user prefences into json object to pass to react
+  const settings = [];
+  const lines = fileToArray("garbo_settings.txt");
+  for (const i in lines) {
+    const data = lines[i].split("\t");
```

Comment:
```suggestion
  for (const line of lines) {
    const [, , name, type, description] = line.split("\t");
```
You want to use for... of to iterate over iterables.

Can you also change the settings file not to have the first two fields? Don't we also want a default value field in that file?

Reply:
I can change the settings file, but fileToArray returns an Object, not an iterable.